src/main/java/org/folio/rest/impl/RequestObjectFactory.java

```diff
@@ -51,32 +61,45 @@
       });
   }
 
-  private CompletableFuture<RequestType> getRequestType(String patronId, String itemId) {
-    final var itemRepository = new ItemRepository(httpClient);
-    final var userRepository = new UserRepository(httpClient);
+  private CompletableFuture<RequestContext> fetchItem(RequestContext requestContext) {
+    return itemRepository.getItem(requestContext.getItemId(), okapiHeaders)
+      .thenApply(requestContext::setItem);
+  }
 
-    CompletableFuture<JsonObject> userFuture = userRepository.getUser(patronId, okapiHeaders);
-    CompletableFuture<JsonObject> itemFuture = itemRepository.getItem(itemId, okapiHeaders);
+  private CompletableFuture<RequestContext> fetchUser(RequestContext requestContext) {
+    return userRepository.getUser(requestContext.getPatronId(), okapiHeaders)
+      .thenApply(requestContext::setUser);
+  }
 
+  private CompletableFuture<RequestContext> getRequestType(RequestContext requestContext) {
     RequestTypeParameters requestTypeParams = new RequestTypeParameters();
```

Comment:
`requestTypeParams `, `requestPolicyId` and `requestPolicy` can be stored inside RequestContext

Reply:
I've tried this approach. Unfortunately, when I store RequestTypeParameters requestTypeParams as a class field when this method is called, vert.x hangs and does not work. Why ? I do not understand.

Reply:
It seems my IDE was broken that day. I've refactored code.